### PR TITLE
Exposing all data of the profile.data on User#profile, renamed userId…

### LINKED ISSUE #3268: JWT metadata missing on User object
## Goals

I have specified metadata fields in the custom JWT authentication settings, but these fields do not appear on the user object after logging in.
I am working with `realm-web` version `0.9.0`

## Expected Results

When using the Stitch SDK these fields could be found on the `user.profile.data` object (`mongodb-stitch-browser-sdk` version `4.8.0`).

## Actual Results

The `user.profile.data` object does not exist. See screenshots.

realmjwt1
realmjwt2

## Steps to Reproduce

Use custom JWT authentication with metadata fields. Login in the browser and check the logged in user object.

## Code Sample

```
const loggedInUser = await realmApp.logIn(Realm.Credentials.jwt(jwtString))
console.log(loggedInUser.profile.data) // undefined
```

## Version of Realm and Tooling

- Realm JS SDK Version: realm-web 0.9.0
- Node or React Native: Web
- Client OS & Version: Chrome 85 on macOS 10.15.7
- Which debugger for React Native: None

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

Merging this PR will:
- Expose all data of the `profile.data` on `User#profile` (Realm Web only - still missing on Realm JS)
- Rename `userId` to `id` on `User#identities` elements (Realm JS and types)
- Expose `providerType` on a `User` (Realm Web).
- Add three properties missing from the RPC User proxy object.

This closes #3268 and a few other untracked issues.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: packages/realm-web/src/UserProfile.ts [hunk 1]
```diff
@@ -65,79 +67,63 @@ const DATA_MAPPING: { [k in DataKey]: keyof UserProfile } = {
...
             }
 
-            if (Array.isArray(response.identities)) {
-                this.identities = response.identities.map((identity: any) => {
+            if (Array.isArray(identities)) {
+                this.identities = identities.map((identity: any) => {
                     return {
                         id: identity.id,
-                        providerId: identity["provider_id"],
+                        // providerId: identity["provider_id"],
```
Comment (kneth) [inline packages/realm-web/src/UserProfile.ts:99]: Should the line be removed?

File: docs/sync.js [hunk 2]
```diff
@@ -547,29 +547,52 @@ class ApiKeyAuth {
...
 }
 
+/**
+ * The type of an authentication provider, used to authenticate a user.
+ * @typedef Realm.App.Sync~ProviderType
+ * @type {("anon-user"|"api-key"|"local-userpass"|"custom-function"|"custom-token"|"oauth2-google"|"oauth2-facebook"|"oauth2-apple")}
+ */
+
+/**
+ * Describes an error when an incompatible synced Realm is opened. The old version of the Realm can be accessed in readonly mode using the configuration() member
```
Comment (kneth) [inline docs/sync.js:557]: This is the description of `IncompatibleSyncedRealmError`. We need a correct description of `UserIdentity`.

**Review by kneth — CHANGES_REQUESTED** [review]

File: docs/sync.js (hunk 2 above)
Comment (kraenhansen) [reply docs/sync.js:557]: Thanks!

File: packages/realm-web/src/UserProfile.ts (hunk 1 above)
Comment (kraenhansen) [reply packages/realm-web/src/UserProfile.ts:99]: Probably 😊 I might have wanted to leave it in as a signal that this is infact also returned from the API, but it should probably be removed.

**Review by fronck — APPROVED** [review]
LGTM

**Review by jsflax — APPROVED** [review]

**Review by cesarvr — APPROVED** [review]

**Review by kneth — APPROVED** [review]

**Pull request merged**